Zendro_requests_with_python.md

```diff
@@ -0,0 +1,246 @@
+
+
+```python
+
+# to make a read request and not a mutation we change mutation to query
+country_cities = """
+    query {
+          readOneCountry(country_id: "JP") {
+            citiesFilter(pagination: { limit:10 }) {
+              name
+            }
+          }
+    } 
+"""
+
+country_cities_response = session.post(
+    "https://zendro.conabio.gob.mx/api/graphql", verify=False, json={"query": country_cities}
+)
+
+country_cities_response.json()
+```
+
+
+
+
+    {'data': {'readOneCountry': {'citiesFilter': [{'name': 'Osaka'},
+        {'name': 'Suwon'}]}}}
+
+
+
+We can **update the entries in our table with mutations, for this we need to specify the id of the entry we are going to update. It is also helpful to retrieve in the response the field we just update so we know the mutation ran correctly.
+
+In the next example we update the field  `population` to correct the population of Japan.
+
+
+```python
+
+# we can make changes to a table's entry using the specific type queries for that
+# in this case we use 'updateCountry' to fix the population of Japan
+update_country = """
+    mutation {
+          updateCountry(
+            country_id: "JP",
+            population: 125000000
+          ) {
+            name
+            population
+          }
+    } 
+"""
+
+update_response = session.post(
+    "https://zendro.conabio.gob.mx/api/graphql", verify=False, json={"query": update_country}
+)
+
+update_response.json()
+```
+
+
+
+
+    {'data': {'updateCountry': {'name': 'Japan', 'population': 125000000}}}
+
+
+
+Finally to **delete** an entry, first we need to disassociate it from its relations, once our entry is no longer related to any other table, we can delete it. 
+
+In the first examples, when we created our cities, we create the city of `Suwon`, which doesn't belong to Japan, so we first disassociate it to the country, and then we can delete it, by typing  its id in the `deleteCity` mutation.
+
+The delete mutations normally just return a string, instead of the fields of the table we are targeting.
+
+
+```python
+
+# to delete an entry, first we need to disassociate the entry of its relations
+delete_city = """
+    mutation {
+          updateCity(
+            city_id: 7,
+            removeCountry: "JP"
+          ) { city_id country_id }
+          deleteCity(
+            city_id: 7 
+          ) 
+    } 
+"""
+# the delete query returns a string
+delete_response = session.post(
+    "https://zendro.conabio.gob.mx/api/graphql", verify=False, json={"query": delete_city}
+)
+
+delete_response.json()
+```
+
+
+
+
+    {'data': {'updateCity': {'city_id': '7', 'country_id': None},
+      'deleteCity': 'Item successfully deleted'}}
+
+
```

Comment:
It might be useful to add an example on how to pass variables to a given query, because that is what a user would usually do: define their query and then pass in the dynamic parameters (search, order, pagination,...) at run time.

Reply:
Do you have an example of this or link to the docs? not sure how to do it.

Reply:
Dear Alicia,
Here is the tutorial for variables from Graphql documentation: https://graphql.org/learn/queries/#variables
And we also have some examples in graphql-server-model-codegen repository.
For example, in the file `graphql-server-model-codegen/test/unit_test_misc/test-describe/cursor-based-pagination.js`, 
we have a query in line 142 as the following:
```
    let query = \`query booksConnection($search: searchBookInput $pagination: paginationCursorInput! $order: [orderBookInput]){
        booksConnection(search:$search pagination:$pagination order:$order){ edges{cursor node{  id  title
          genre
          publisher_id
         } } pageInfo{ startCursor endCursor hasPreviousPage hasNextPage } } }\`
```
Here we define three variables, namely search, pagination and order. User can pass these variables dynamically.
And we can send query and variables via axios:
```
      let response = await axios.post(
        remoteZendroURL, 
        {
          query: query,
          variables: {search: search, order:order, pagination: pagination},
        },
        opts
      );
```

Reply:
I added a short intro in GraphQL_intro.md and a more detailed example in Zendro_requests_with_python.md. I didn't used our cities-rivers examples because our instance seems to be down and I wan't to get this done today. But I don't think it is necessary to have the city-rivers example for this, so the examples I added and the link to the documentation should be enough. Commit 55ba0f9